### PR TITLE
Feature/dapp quick pay route

### LINKED ISSUE #2882: dApp - quick pay feature
The dApp has a feature which allows to create links that include parameters for a full payment (token address, target address, amount and payment identifier). With the current redirect functionality, this will navigate the user automatically to a transfer after he connected to the dApp. Then he either does a direct payment or has to find a route first.

While this is working, it isn't a full mature feature. It doesn't work reliable for many cases, plus it misses some useful additions.

One missing feature is the complete communication cycle. This means that other applications can create links for a user to easily make a specific payment. But once the payment is done, the user must return manually to the origin application. Furthermore does the origin application has no clue that the payment is done.

The major issue is the unreliable user flow. The current feature only works for users which have used the dApp already and have a setup connection to the specific token network the payment targets. If these requirements are not given, the user will be dropped into an on-boarding flow or a connect token flow. The issue that in the meantime the dApp forgot that the user came originally to do this specific payment.
But also if all the previous made requirements are met, it could still fail. If the user has not already a direct channel with the payment target node, he has to find a payment route via a PFS and then do a mediated transfer. So far a normal payment flow as usual. But if the PFS can't find a route in the Raiden network, the payment flow of the user will fail. The flow basically stops. He will return to the default screen and that's it. He would now have to manually open a direct channel to the target or maybe deposit more tokens into his already open channel (for the case this is the issue to find a route). But again, the dApp fully lost the state of the original intention to do this payment and return to the other application.

Having this "pay and go" feature fully working is a key to improve the usability of Raiden/the dApp and to support new projects like the Raiden PayWall.

### PR DESCRIPTION
**Pull request by weilbith**
**Thank you for submitting this pull request :)**

Fixes #2882

**Short description**
The most exciting feature for the dApp since the connection manager 😬 
All the previous refactoring enabled to do such nice thing like this.

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [x] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1. Please checkout the image with the different cases in the linked issue.

### CODE REVIEW

**Comment by weilbith** [comment]
I need to add a CHANGELOG entry.

File: raiden-dapp/src/locales/en.json
Comment (taleldayekh) [inline raiden-dapp/src/locales/en.json:278]: "Enter an amount larger than {funds}"

File: raiden-dapp/src/locales/en.json
Comment (taleldayekh) [inline raiden-dapp/src/locales/en.json:776]: "The target is online but does not seem to receive payments."

File: raiden-dapp/src/locales/en.json
Comment (taleldayekh) [inline raiden-dapp/src/locales/en.json:777]: "Please contact the payment target to make sure they can receive payments and try again."

File: raiden-dapp/src/locales/en.json
Comment (taleldayekh) [inline raiden-dapp/src/locales/en.json:312]: "Please wait"

File: raiden-dapp/src/locales/en.json
Comment (taleldayekh) [inline raiden-dapp/src/locales/en.json:317]: "Deposit and Pay"

File: raiden-dapp/src/locales/en.json
Comment (taleldayekh) [inline raiden-dapp/src/locales/en.json:330]: "confirm-button-label": "Open Channel and Pay"

File: raiden-dapp/src/locales/en.json
Comment (taleldayekh) [inline raiden-dapp/src/locales/en.json:888]: "Channel Deposit and Pay"

File: raiden-dapp/src/locales/en.json
Comment (taleldayekh) [inline raiden-dapp/src/locales/en.json:889]: "Channel Open and Pay"

File: raiden-dapp/src/locales/en.json
Comment (taleldayekh) [inline raiden-dapp/src/locales/en.json]: "Some payment parameters are invalid. Please verify that the link is correct."

File: raiden-dapp/src/views/QuickPayRoute.vue
Comment (taleldayekh) [inline raiden-dapp/src/views/QuickPayRoute.vue:235]: Perhaps use `switch` here instead.

File: raiden-dapp/src/views/QuickPayRoute.vue
Comment (taleldayekh) [inline raiden-dapp/src/views/QuickPayRoute.vue:245]: Perhaps use `switch` here instead.

**Review by taleldayekh — APPROVED** [review]
Went through this together with @weilbith and looks very good, thx! 👍🏼

File: raiden-dapp/src/views/QuickPayRoute.vue
Comment (weilbith) [reply raiden-dapp/src/views/QuickPayRoute.vue]: ✔️

File: raiden-dapp/src/views/QuickPayRoute.vue
Comment (weilbith) [reply raiden-dapp/src/views/QuickPayRoute.vue]: ✔️

File: raiden-dapp/src/locales/en.json
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

File: raiden-dapp/src/locales/en.json
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

File: raiden-dapp/src/locales/en.json
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

File: raiden-dapp/src/locales/en.json
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

File: raiden-dapp/src/locales/en.json
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

File: raiden-dapp/src/locales/en.json
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

File: raiden-dapp/src/locales/en.json
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

File: raiden-dapp/src/locales/en.json
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

File: raiden-dapp/src/locales/en.json
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

**Comment by weilbith** [comment]
@taleldayekh thanks for the nice review. I always appreciate that you take the time to do it together and discuss the details. Also thanks to always improve my user translations! 🙏🏾

**Comment by weilbith** [comment]
I addressed all feedback. I'm very positive that I don't need an in-between CI confirmation. Therefore I'll go ahead and squash the fixups. Then I'll wait for CI to finally merge it.

**Comment by weilbith** [comment]
Sorry, I messed something up... So much to I'm optimistic.
We really need some basic commit/push hooks.

**Comment by andrevmatos** [comment]
@weilbith can you rebase on top of master on next run, please? some dependencies updates (including jest/test related stuff) got merged. Hopefully it won't impact actual code nor conflict, but it'll be better to have that tested before merging. Thanks

**Comment by weilbith** [comment]
Rebase as requested by Andre...

**Comment by weilbith** [comment]
Okay, apparently this did break something...

**Comment by weilbith** [comment]
The Jest update changed the behavior of the `toContain` has changed.

**Comment by weilbith** [comment]
I don't know why the `license/cla` check is not reporting. But I'm the only one who contributed code to this PR. And I have signed. So I use my permissions to merge this PR anyway.

**Pull request merged**
